### PR TITLE
Fix SubnetPort allocate address for DHCP

### PR DESCRIPTION
**Pull request by yanjunz97**

### CODE REVIEW

File: pkg/nsx/services/subnetport/builder.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
 	controllercommon "github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	util2 "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
```
Comment (wenqiq) [inline pkg/nsx/services/subnetport/builder.go:20]: util2-->nsxutil. The package alias name should indicate the purpose or function of the package.

File: pkg/nsx/util/utils.go [hunk 2]
```diff
@@ -708,3 +708,10 @@ func IsInvalidLicense(err error) bool {
 	}
 	return invalidLicense
 }
+
+func ParseDHCPMode(mode string) string {
+	// Transfer DHCPDeactivated to DHCP_DEACTIVATED
+	nsxMode := strings.ToUpper(mode)
+	nsxMode = nsxMode[:4] + "_" + nsxMode[4:]
```
Comment (wenqiq) [inline pkg/nsx/util/utils.go:715]: Should add a length check (len(mode) < 5) to prevent slicing errors? BTW, had better a unit-test for this function.

**Review by wenqiq — APPROVED** [review]
lgtm

**Review by zhengxiexie — APPROVED** [review]
lgtm

**Pull request merged**